From b38adf6ca4c589a7a3fc510414dff10d5de6b6b3 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 7 Feb 2013 15:27:48 +0800
Subject: [PATCH 387/505] ENGR00243850 V4L2:Correct output buffer field before
 queue it

Since MXC V4L2 output driver changes to get every buffer's
field, we need to provide buffer field information before
queue it, otherwise, interlaced frames cannot be de-interlaced.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
---
 test/mxc_v4l2_test/mxc_v4l2_output.c | 3 ++-
 test/mxc_v4l2_test/mxc_v4l2_tvin.c   | 3 +++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/test/mxc_v4l2_test/mxc_v4l2_output.c b/test/mxc_v4l2_test/mxc_v4l2_output.c
index ddf6513..14788e4 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_output.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_output.c
@@ -327,7 +327,8 @@ mxc_v4l_output_test(FILE *in)
 
                 buf.timestamp.tv_sec = tv_start.tv_sec;
                 buf.timestamp.tv_usec = tv_start.tv_usec + (g_frame_period * i);
-
+		if (g_vdi_enable)
+			buf.field = V4L2_FIELD_INTERLACED_TB;
                 if ((retval = ioctl(fd_v4l, VIDIOC_QBUF, &buf)) < 0)
                 {
                         printf("VIDIOC_QBUF failed %d\n", retval);
diff --git a/test/mxc_v4l2_test/mxc_v4l2_tvin.c b/test/mxc_v4l2_test/mxc_v4l2_tvin.c
index 06b1437..a54a40b 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_tvin.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_tvin.c
@@ -584,6 +584,9 @@ next:
 
 		output_buf.timestamp.tv_sec = tv_start.tv_sec;
 		output_buf.timestamp.tv_usec = tv_start.tv_usec + (g_frame_period * i);
+		if (g_vdi_enable)
+			output_buf.field = g_tb ? V4L2_FIELD_INTERLACED_TB :
+						  V4L2_FIELD_INTERLACED_BT;
 		if (ioctl(fd_output_v4l, VIDIOC_QBUF, &output_buf) < 0)
 		{
 			printf("VIDIOC_QBUF failed\n");
-- 
1.8.0

